pkg/smi/health.go

```diff
@@ -0,0 +1,82 @@
+package smi
+
+import (
+	"context"
+
+	extensionsClientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+
+	"github.com/openservicemesh/osm/pkg/version"
+)
+
+// HealthChecker has SMI kube config to access SMI CRDS
+type HealthChecker struct {
+	SMIKubeConfig *rest.Config
+}
+
+// Liveness is the Kubernetes liveness probe handler.
+func (smi HealthChecker) Liveness() bool {
+	return true
+}
+
+// Readiness is the Kubernetes readiness probe handler.
+func (smi HealthChecker) Readiness() bool {
+	return checkSMICrdsExist(smi.SMIKubeConfig)
+}
+
+// GetID returns the ID of the probe
+func (smi HealthChecker) GetID() string {
+	return "SMI_CRDS_ExistenceChecker"
```

Comment:
Just SMI should suffice here.
```suggestion
	return "SMI"
```